### PR TITLE
fix(aws-amplify-react): confirmSignIn bugfix

### PR DESCRIPTION
**Pull request by haverchuck**
*Description of changes:*
Adds e.preventDefault to confirm function in ConfirmSignIn component to prevent page reload

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/aws-amplify-react/src/Auth/ConfirmSignIn.jsx [hunk 1]
```diff
@@ -63,7 +63,8 @@ export default class ConfirmSignIn extends AuthPiece {
             });
     }
 
-    confirm() {
+    confirm(e) {
```
Comment (sammartinez) [inline packages/aws-amplify-react/src/Auth/ConfirmSignIn.jsx]: Can we change `e` to `event`, it makes it clearer as we do use `e` for catch blocks in some of our code.

Comment (elorzafe) [reply packages/aws-amplify-react/src/Auth/ConfirmSignIn.jsx]: +1

Comment (haverchuck) [reply packages/aws-amplify-react/src/Auth/ConfirmSignIn.jsx]: Done!

Comment (sammartinez) [reply packages/aws-amplify-react/src/Auth/ConfirmSignIn.jsx]: Thanks!

**Pull request merged**